### ISSUE TITLE
Log `--approve` warning for `update cluster` command

### ISSUE THREAD

**Issue by martina-if** [weaveworks/eksctl#1643]
## Description

Today when `eksctl update cluster` command is run, the user needs to add the flag `--approve` to have eksctl actually perform the upgrade. That's because eksctl acts in "planning" mode. This is logged very subtlety:

```console
$ eksctl update cluster $CLUSTER_NAME
[ℹ]  eksctl version fb1e0f53d42f6ee88b56d6bb0e14ab2c2465fc36
[ℹ]  using region ap-northeast-1
[ℹ]  re-building cluster stack "eksctl-martina-test-f16-cluster"
[ℹ]  (plan) updating stack to add new resources [FargatePodExecutionRole] and outputs [FargatePodExecutionRoleARN]
[ℹ]  checking security group configuration for all nodegroups
[ℹ]  all nodegroups have up-to-date configuration
```

The key is in the log line number 4 `(plan) ...`.

## Expected

I think this command should print a warning as the last log line like in the case of deleting a nodegroup:

```console
$ eksctl delete nodegroup -f fargate-managed-normal.yaml --include ng-2
[ℹ]  eksctl version 0.2.0-1019-gdc6f0367
[ℹ]  using region ap-northeast-1
[ℹ]  comparing 2 nodegroups defined in the given config ("fargate-managed-normal.yaml") against remote state
[ℹ]  combined include rules: ng-2
[ℹ]  1 nodegroup (ng-2) was included (based on the include/exclude rules)
[ℹ]  2 nodegroups (managed-1, ng-1) were excluded (based on the include/exclude rules)
[ℹ]  (plan) would delete 1 nodegroups from auth ConfigMap in cluster "martina-test-all"
[ℹ]  (plan) would drain 1 nodegroup(s) in cluster "martina-test-all"
[ℹ]  (plan) would delete 1 nodegroups from cluster "martina-test-all"
[ℹ]  (plan) 1 task: { delete nodegroup "ng-2" [async] }
[✔]  (plan) would have deleted 1 nodegroup(s) from cluster "martina-test-all"
[!]  no changes were applied, run again with '--approve' to apply the changes
```

So it would look like:

```console
$ eksctl update cluster $CLUSTER_NAME
[ℹ]  eksctl version fb1e0f53d42f6ee88b56d6bb0e14ab2c2465fc36
[ℹ]  using region ap-northeast-1
[ℹ]  re-building cluster stack "eksctl-martina-test-f16-cluster"
[ℹ]  (plan) updating stack to add new resources [FargatePodExecutionRole] and outputs [FargatePodExecutionRoleARN]
[ℹ]  checking security group configuration for all nodegroups
[ℹ]  all nodegroups have up-to-date configuration
[!]  no changes were applied, run again with '--approve' to apply the changes
```

**Comment by spaghettifunk** [comment]
Hi @martina-if 

I was taking a look at this issue, and I noticed that the reason why the `[!]  no changes were applied, run again with '--approve' to apply the changes` is not printed is due to the condition in this [line](https://github.com/weaveworks/eksctl/blob/master/pkg/ctl/update/cluster.go#L135).
The 'updated' command has the wanted behavior, if we loosen up the condition for printing the message. 

However, I was wondering if this solution makes sense. It seems to me that the `cmd.Plan` when set to `true`, has "priority" on any other condition. If the `--approve` flag is not specified, then no matter if `stackUpdateRequired = true` or `versionUpdateRequired = true`,  nothing will be `updated` on the cluster. 

Long story short: instead of `cmdutils.LogPlanModeWarning(cmd.Plan && (stackUpdateRequired || versionUpdateRequired))`, we can write this `cmdutils.LogPlanModeWarning(cmd.Plan)`. Does this make sense?
